pkg/logcli/query/query.go

```diff
@@ -80,76 +79,76 @@ func (q *Query) DoQuery(c client.Client, out output.LogOutput, statistics bool)
 			q.printStats(resp.Data.Statistics)
 		}
 		_, _ = q.printResult(resp.Data.Result, out, nil)
-	} else {
-		if q.Limit < q.BatchSize {
-			q.BatchSize = q.Limit
-		}
-		resultLength := q.BatchSize
-		total := 0
-		start := q.Start
-		end := q.End
-		var lastEntry []*loghttp.Entry
-		for total < q.Limit {
-			bs := q.BatchSize
-			// We want to truncate the batch size if the remaining number
-			// of items needed to reach the limit is less than the batch size
-			if q.Limit-total < q.BatchSize {
-				// Truncated batchsize is q.Limit - total, however we add to this
-				// the length of the overlap from the last query to make sure we get the
-				// correct amount of new logs knowing there will be some overlapping logs returned.
-				bs = q.Limit - total + len(lastEntry)
-			}
-			resp, err = c.QueryRange(q.QueryString, bs, start, end, d, q.Step, q.Interval, q.Quiet)
-			if err != nil {
-				log.Fatalf("Query failed: %+v", err)
-			}
+		return
+	}
 
-			if statistics {
-				q.printStats(resp.Data.Statistics)
-			}
+	if q.Limit < q.BatchSize {
+		q.BatchSize = q.Limit
+	}
+	resultLength := q.BatchSize
```

Comment:
I think this was inefficient assignment.
```suggestion
	resultLength := 0
```

Reply:
@cyriltovena is resultLength in line 88 depends to q.BatchSize in line 86.